WcaOnRails/app/assets/javascripts/currencies.js

```diff
@@ -0,0 +1,77 @@
+window.wca = window.wca || {};
+
+wca.getCurrencyInfo = function(isoCode) {
+  return wca._currenciesInfo[isoCode] || wca._currenciesInfo.USD;
+};
+
+// Create a mask for an amount input in the given currency, using autoNumeric
+// 'action' can be either "init" or "update"
+// 'elemId' is a selector for the targeted input field
+wca.applyCurrencyMask = function(action, elemId, currencyIsoCode) {
+  if (!$(elemId).length)
+    return;
+  var entry = wca.getCurrencyInfo(currencyIsoCode);
+  var currentVal = 0;
+
+  // Get current val
+  if (action == "update") {
+    currentVal = wca.getValueInCurrency(elemId);
+  } else if (action == "init") {
+    currentVal = $(elemId).val();
+  } else {
+    return;
+  }
+
+  // Reconfigure
+  $.data($(elemId)[0], "current_subunit_to_unit", entry.subunit_to_unit);
+  $(elemId).autoNumeric(action, {
+    currencySymbol: entry.symbol,
+    currencySymbolPlacement: (entry.symbol_first)?'p':'s',
```

Comment:
total nit, but someday we may set up a js style checker: spacing seems to be missing here, and therefore inconsisten with the ternary a few lines lower.